src/prettytable/prettytable.py

```diff
@@ -1583,14 +1583,27 @@ def _compute_widths(self, rows, options):
                 title_width = 0
             min_table_width = self.min_table_width or 0
             min_width = max(title_width, min_table_width)
-            table_width = self._compute_table_width(options)
-            if table_width < min_width:
-                # To make enough space for the title, all columns have to
-                # grow enough that their combined widths (and the borders
-                # between them) are equal to the length of the title.
-                border_char_count = 3 * (len(widths) - 1)
-                scale = 1.0 * (len(options["title"]) - border_char_count) / sum(widths)
-                widths = [int(math.ceil(w * scale)) for w in widths]
+            if options["border"]:
+                borders = len(widths) + 1
+            else:
+                if options["preserve_internal_border"]:
+                    borders = len(widths)
+                else:
+                    borders = 0
```

Comment:
Can we simplify this?

```suggestion
            if options["border"]:
                borders = len(widths) + 1
            elif options["preserve_internal_border"]:
                borders = len(widths)
            else:
                borders = 0
```